api/v1beta2/awsmachine_webhook_test.go

```diff
@@ -217,6 +217,38 @@ func TestAWSMachineCreate(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "invalid case, MarketType set to MarketTypeCapacityBlock and spotMarketOptions are specified",
+			machine: &AWSMachine{
+				Spec: AWSMachineSpec{
+					MarketType:        MarketTypeCapacityBlock,
+					SpotMarketOptions: &SpotMarketOptions{},
+					InstanceType:      "test",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid case, MarketType set to MarketTypeOnDemand and spotMarketOptions are specified",
+			machine: &AWSMachine{
+				Spec: AWSMachineSpec{
+					MarketType:        MarketTypeOnDemand,
+					SpotMarketOptions: &SpotMarketOptions{},
+					InstanceType:      "test",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid MarketType set to MarketTypeCapacityBlock is specified",
+			machine: &AWSMachine{
+				Spec: AWSMachineSpec{
+					MarketType:   MarketTypeCapacityBlock,
+					InstanceType: "test",
+				},
+			},
+			wantErr: false,
```

Comment:
I think this should be an error because `CapacityReservationID` isn't filled. If I'm right, an additional positive/valid case should also be added.

Reply:
I missed validating the `capacityReservationId` in the webhook when the marketType is set to CapacityBlock. Instead, I raised the error in the `getInstanceMarketOptions` method. It would be better to include the validation in the webhook as well. Thanks

Reply:
fixed